cachito/workers/scm.py

```diff
@@ -181,7 +196,12 @@ def fetch_source(self):
         )
         for previous_archive in previous_archives:
             try:
-                self.update_and_archive(previous_archive)
+                # Excluding previous archives with submodules since there is
+                # no straight-forward way to get rid of the previously included
+                # submodules
+                if "-with-submodules" in str(previous_archive):
+                    continue
```

Comment:
Also, would it be possible to re-use the git history for the submodules as we do with the main repositories? In other words, in case the request includes the submodules pkg manager, could we prefer an archive with submodules?

I'd be OK leaving this for a separate PR, but we would keep consistency on how Cachito fetches sources by doing so (and would pull less data)

Reply:
We tried to achieve that only at first but it was not possible as I had mentioned above.

> Notes:
> Updating a git submodule seems to be very tricky.
> 
> * We would probably not have the previous ref for the submodule just by traversing through the repo from `previous_archive`.
> * Also, if a submodule was previously part of the `previous_archive`, trying to re-clone it seems to be resulting into an error (already exists).
> * There is no obvious way to find out if a submodule was part of the `previous_archive` and hence, there is no straightforward way to implement that check (except probably a hacky way to try to check the directory for empty content)
> * if a previous_archive with submodules is being used, then the subsequent requests don't get rid of the associated submodules. This could lead to dependency issues in the next part of the story.
>   Hence, for the reasons stated above, the best option seems to be to exclude the previous_archives with submodules while updating and archiving repo for a new Cachito request.

Hence, we ended up adding these conditions.

If we really want to achieve that, there should be a spike created to investigate the feasibility first.